### PR TITLE
[feat] LegacyNamespace extended attributes support on IDL interfaces (Resolves #283)

### LINKED ISSUE #283: WebAssembly related types are not properly annotated / namespaced.
The Webassembly API   should be namespaced  by ```WebAssembly```   but several of  the extension  types are not properly annotated  so they try to find the various  type constructors in the global namespace,

 For example the following source for  ```WebAssembly.Memory``` should be annotated as ```@JS('WebAssembly.Memory')```

https://github.com/dart-lang/web/blob/4996dc2acd30ff06f7e500ec76fde8a66db82c0c/lib/src/dom/wasm_js_api.dart#L105-L109

Other likely affected interop types are:

- WebAssembly.CompileError
- WebAssembly.Exception
- WebAssembly.Global
- WebAssembly.Instance
- WebAssembly.LinkError
- WebAssembly.Memory
- WebAssembly.Module
- WebAssembly.RuntimeError
- WebAssembly.Table
- WebAssembly.Tag

Also,  I wonder if  these should / could actually be namespaced in dart in some way,   I was just refactoring some code to try and use dart:js_interop and package:web, in my original implementation I used names like ```WebAssemblyGlobal``` and ```WebAssemblyMemory``` since  types like Memory, Tag, Table etc are pretty generic names and caused a bit of a pain to not conflict with other imports.  Not a huge problem, but it did help when reading code.

### PR DESCRIPTION
**Pull request by rutvik110**
[feat] LegacyNamespace extended attributes support on IDL interfaces (Resolves #283)

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: web/lib/src/dom/wasm_js_api.dart [hunk 1]
```diff
@@ -73,6 +73,7 @@ extension type ModuleImportDescriptor._(JSObject _) implements JSObject {
   external ImportExportKind get kind;
   external set kind(ImportExportKind value);
 }
+@JS('WebAssembly.Module')
```
Comment (kevmoo) [inline web/lib/src/dom/wasm_js_api.dart:76]: Need to put an entry in the `pkg:web` changelog, too!

Comment (rutvik110) [reply web/lib/src/dom/wasm_js_api.dart:76]: Done. Added a bit more condensed description of changes for `pkg:web` changelog,

```Adding a `JS(LegacyNamespace.$extensionTypeName)` annotation on generated `JS` 
  objects if they have an IDL extended attribute `[LegacyNamespace=Foo]` defined 
  in their IDL description.```

File: web/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@
 - Deprecated `TouchListWrapper` and `TouchListConvert` in favor of `JSImmutableListWrapper`.
 - Added `[]` and `[]=` overloaded operators to types which define unnamed
   `getter`s and `setter`s, respectively.
+- Adding a `JS(LegacyNamespace.$extensionTypeName)` annotation on generated `JS`
```
Comment (kevmoo) [inline web/CHANGELOG.md:10]:
```suggestion
- Correctly namespace `WebAssembly` types.
```

File: web/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,9 @@
 - Deprecated `TouchListWrapper` and `TouchListConvert` in favor of `JSImmutableListWrapper`.
 - Added `[]` and `[]=` overloaded operators to types which define unnamed
   `getter`s and `setter`s, respectively.
+- Adding a `JS(LegacyNamespace.$extensionTypeName)` annotation on generated `JS` 
+  objects if they have an IDL extended attribute `[LegacyNamespace=Foo]` defined 
+  in their IDL description.
```
Comment (kevmoo) [inline web/CHANGELOG.md:12]:
```suggestion
- Correctly namespace `WebAssembly` types.
```

This describes the user-visible change.

**Comment by kevmoo** [comment]
@srujzs should review, but this looks great. Thanks so much! 🙏

File: web/CHANGELOG.md (hunk 3 above)
Comment (rutvik110) [reply web/CHANGELOG.md:12]: Neat!

File: web_generator/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,11 @@
 ## 1.0.0-wip
 
-- Initial separation of `web_generator` from `web`.
\ No newline at end of file
+- Initial separation of `web_generator` from `web`.
+- New IDL interface `RHL` added. `ExtendedAttribute` idl interface updated to
```
Comment (srujzs) [inline web_generator/CHANGELOG.md:4]: nit: I think it's okay to not include entries here, but this is okay too. This package is never published, so it's purely for internal purposes.

**Review by srujzs — APPROVED** [review]
Looks great, thank you!

File: web_generator/CHANGELOG.md (hunk 4 above)
Comment (rutvik110) [reply web_generator/CHANGELOG.md:4]: I think it's nice to have this changelog updated with new updates. But I'm good either way. lmk if you want to remove this.

Comment (srujzs) [reply web_generator/CHANGELOG.md:4]: Nope, that's okay - we can keep it. :) LGTM still!

File: web_generator/lib/src/translator.dart [hunk 5]
```diff
@@ -5,6 +5,7 @@
 import 'dart:js_interop';
```
Comment (srujzs) [inline web_generator/lib/src/translator.dart:5]: It looks like Dart CI wants this file formatted FYI.

Comment (rutvik110) [reply web_generator/lib/src/translator.dart:5]: Updated.

**Pull request merged**